src/client/dfuse/pil4dfs/int_dfs.c

```diff
@@ -1017,6 +1163,24 @@ consume_low_fd(void)
 			DS_ERROR(errno, "failed to reserve a low fd");
 			goto err;
 		} else if (low_fd_list[low_fd_count] >= DAOS_MIN_FD) {
+			/* reserve fd 255 too if unused. 255 is commonly used by bash. */
+			fd_dup = fcntl(low_fd_list[0], F_DUPFD, 255);
+			if (fd_dup == -1) {
+				DS_ERROR(errno, "fcntl() failed");
+				goto err;
+			}
+			/* If fd 255 is used, lowest available fd is assigned to fd_dup. */
+			if (fd_dup >= 0 && fd_dup != 255)
+				libc_close(fd_dup);
```

Comment:
That's better.  Are these fds closed at shutdown, if 255 is already in use do you need to mark this so that it won't be closed later.

Reply:
If fd 255 is opened by libpil4dfs, it will not be closed explicitly at process exit.
Are you suggesting we actively avoid closing fd 255 inside close()?
We could actually close it then reopen fd 255 immediately to reserve it as much as possible.

Reply:
I don't know why pil4dfs wants to reserve fd 255 but if it does open it then it should close it on exit.

Reply:
ok. I added flags to track whether fd 255 and DAOS_DUMMY_FD are reserved by current process. If yes, they are closed on exit. Thank you!